vcxagency-node/src/service/storage/storage.js

```diff
@@ -146,21 +168,22 @@ async function createDataStorage (appStorageConfig) {
     filterAgentConnDids = assureValidFilter(filterAgentConnDids)
     filterUids = assureValidFilter(filterUids)
     filterStatusCodes = assureValidFilter(filterStatusCodes)
-    let insertQuery = 'SELECT * from messages WHERE agent_did = ?'
+    let query = 'SELECT * from messages WHERE agent_did = ?'
     const values = [agentDid]
     if (filterAgentConnDids.length > 0) {
       values.push(filterAgentConnDids)
-      insertQuery += ' AND agent_connection_did IN(?)'
+      query += ' AND agent_connection_did IN(?)'
     }
     if (filterUids.length > 0) {
       values.push(filterUids)
-      insertQuery += ' AND uid IN(?)'
+      query += ' AND uid IN(?)'
```

Comment:
Let's put `uid` on the second place in the query to make sure the index is used.